docs/src/api.md

```diff
@@ -0,0 +1,46 @@
+# AMDGPUnative API Reference
+
+## Kernel launching
+
+```@docs
+@roc
+AMDGPUnative.AbstractKernel
+AMDGPUnative.rocfunction
+```
+
+## Device code API
+
+### Thread indexing
+
+#### HSA nomenclature
+
+```@docs
+workitemIdx
+workgroupIdx
+workgroupDim
+gridDim
+gridDimWG
+```
+
+#### CUDA nomenclature
+
+Use these functions for compatibility with CUDAnative.jl.
+
+```@docs
+threadIdx
+blockIdx
+blockDim
+```
+
+### Synchronization
+
+```@docs
+sync_workgroup
+```
+
+### Other
+
+```@docs
+AMDGPUnative.dynamic_rocfunction
```

Comment:
`dynamic_rocfunction` isn't yet wired up, so I would probably remove it from here for now.